test/libsolidity/syntaxTests/experimental/builtin/builtin_type_definition.sol

```diff
@@ -0,0 +1,88 @@
+pragma experimental solidity;
+
+type void = __builtin("void");
+
+type bool = __builtin("bool");
+type word = __builtin("word");
+type integer = __builtin("integer");
+type unit = __builtin("unit");
+
+type fun(T, U) = __builtin("fun");
+type pair(T, U) = __builtin("pair");
```

Comment:
Since this is a built-in and the type variables will never be used for anything, maybe it would be better to omit them in the definition?

```solidity
type fun = __builtin("fun");
```

I could also do this, just to keep them connected to something:
```solidity
type fun(T, U) = __builtin("fun", T, U);
```
not sure if it makes sense though.